compiler/src/main/scala/org/scalajs/nscplugin/TypeConversions.scala

```diff
@@ -0,0 +1,173 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.nscplugin
+
+import scala.tools.nsc._
+
+import org.scalajs.ir
+import ir.{Definitions, Types}
+
+/** Conversions from scalac `Type`s to the IR `Type`s and `TypeRef`s. */
+trait TypeConversions extends SubComponent { this: GenJSCode =>
+  import global._
+  import definitions._
+
+  def toIRType(t: Type): Types.Type =
+    convert(t, 0)(ToIRTypeFinisher)
+
+  def toTypeRef(t: Type): Types.TypeRef =
+    convert(t, 0)(ToTypeRefFinisher)
+
+  private sealed abstract class ConversionFinisher {
```

Comment:
Instead of this, consider having `convert` return a `(Symbol, Int)` and do the finishing at the call site. This will make for much simpler code IMO.

Reply:
I added a commit with that suggestion. If you prefer that one, I'll squash the commits.

Reply:
I very much prefer that one :) Thanks.